gpcontrib/arenadata_toolkit/src/arenadata_toolkit_guc.c

```diff
@@ -0,0 +1,274 @@
+			return false;
+
+		return true;
+	}
+
+	*manual = true;
+	return false;
+}
+
+/*
+ * Prohibit changing the arenadata_toolkit.tracking_is_db_tracked value manually
+ */
+static bool
+check_tracked(bool *newval, void **extra, GucSource source)
+{
+	bool		manual = false;
+
+	if (check_guc(&is_tracked_unlocked, source, &manual))
+		return true;
+
+	if (manual)
+		GUC_check_errmsg("cannot change tracking status outside the tracking_register_db function");
+	return false;
+}
+
+/*
+ * Prohibit changing the arenadata_toolkit.tracking_snapshot_on_recovery value manually
+ */
+static bool
+check_get_full_snapshot_on_recovery(bool *newval, void **extra, GucSource source)
+{
+	bool		manual = false;
+
+	if (check_guc(&is_get_full_snapshot_on_recovery_unlocked, source, &manual))
+		return true;
+
+	if (manual)
+		GUC_check_errmsg("cannot change tracking status outside the tracking_set_snapshot_on_recovery function");
+	return false;
+}
+
+/*
+ * Prohibit changing the arenadata_toolkit.tracking_relkinds value manually
+ */
+static bool
+check_relkinds(char **newval, void **extra, GucSource source)
+{
+	bool		manual = false;
+
+	if (check_guc(&is_relkinds_unlocked, source, &manual))
+		return true;
+
+	if (manual)
+		GUC_check_errmsg("cannot change tracking status outside the tracking_register_relkinds function");
+	return false;
+}
+
+/*
+ * Prohibit changing the arenadata_toolkit.tracking_schemas value manually
+ */
+static bool
+check_schemas(char **newval, void **extra, GucSource source)
+{
+	bool		manual = false;
+
+	if (check_guc(&is_schemas_unlocked, source, &manual))
+		return true;
+
+	if (manual)
+		GUC_check_errmsg("cannot change tracking status outside the tracking_register_schema function");
+	return false;
+}
+
+/*
+ * Prohibit changing the arenadata_toolkit.tracking_relstorages value manually
+ */
+static bool
+check_relstorages(char **newval, void **extra, GucSource source)
+{
+	bool		manual = false;
+
+	if (check_guc(&is_relstorages_unlocked, source, &manual))
+		return true;
+
+	if (manual)
+		GUC_check_errmsg("cannot change tracking status outside the tracking_register_relstorages function");
+	return false;
+}
+
+void
+tf_guc_define(void)
+{
+	DefineCustomIntVariable("arenadata_toolkit.tracking_bloom_size",
+				   "Size of bloom filter in bytes for each tracked database",
+							NULL,
+							&bloom_size,
+							DEFAULT_BLOOM_SIZE_BYTES,
+							MIN_BLOOM_SIZE_BYTES,
+							MAX_BLOOM_SIZE_BYTES,
+							PGC_POSTMASTER,
+							0,
+							NULL,
+							NULL,
+							NULL
+		);
+
+	DefineCustomIntVariable("arenadata_toolkit.tracking_db_track_count",
+							"Count of tracked databases.",
+							NULL,
+							&db_track_count,
+							DEFAULT_DB_TRACK_COUNT,
+							MIN_DB_TRACK_COUNT,
+							MAX_DB_TRACK_COUNT,
+							PGC_POSTMASTER,
+							0,
+							NULL,
+							NULL,
+							NULL
+		);
+
+	DefineCustomBoolVariable("arenadata_toolkit.tracking_is_db_tracked",
+							 "Is current database tracked.",
```

Comment:
Is it expected result (`tracking_is_db_tracked` is showing `off` while `tracking_register_db()` was called)?
```
postgres=# select arenadata_toolkit.tracking_register_db();
 tracking_register_db 
----------------------
 t
(1 row)

postgres=# 
postgres=# show arenadata_toolkit.tracking_is_db_tracked;
 arenadata_toolkit.tracking_is_db_tracked 
------------------------------------------
 off
(1 row)
```

Reply:
fixed